### PR TITLE
Definitions for node-siege

### PR DESCRIPTION
**Pull request by Goldsmith42**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/node-siege/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Type definitions for node-siege 0.0
+// Project: https://github.com/npr/node-siege
+// Definitions by: Zlatko Andonovski <https://github.com/Goldsmith42>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { EventEmitter } from 'events';
+
+declare class Siege extends EventEmitter {
+    args(args: string): Siege;
+    run(): Siege;
```
Comment (peterblazejewicz) [inline types/node-siege/index.d.ts:10]: Run requires args, you can use TS var args here:
https://github.com/npr/node-siege/blob/master/lib/siege.js#L26
either `any[]` or generic ones:
https://www.typescriptlang.org/docs/handbook/generics.html#hello-world-of-generics

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Goldsmith42 thx!

File: types/node-siege/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    validateArgs(): void;
+    error(err: any): void;
```
Comment (DanielRosenwasser) [inline types/node-siege/index.d.ts:12]: This seems to be missing methods from the README https://www.npmjs.com/package/siege

File: types/node-siege/node-siege-tests.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    .args('-t5s -c 10 -b -f./urls.txt')
+    .on('stdout', (data) => {})
+    .on('stderr', (data) => {})
+    .on('error', (err) => {})
+    .on('exit', () => {
+        if (attack.running) {
+            throw new Error('Exit received but still marked as running');
+        }
+    })
+    .run();
```
Comment (DanielRosenwasser) [inline types/node-siege/node-siege-tests.ts:13]: Can you add tests like

```ts
siege()
  .on(3000)
  .for(10000).times
  .get('/')
  .attack()
```

```ts
siege('/dir/name/app.js')
  .wait(1000)
  .get('/')
  .attack()
```

```ts
siege()
  .withCookie
  .post('/login', {user: pass}).for(1).times
  .get('/')
  .attack()
```

```
siege()
  .concurrent(100)
  .get('/')
  .attack()
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/node-siege/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Type definitions for node-siege 0.0
```
Comment (DanielRosenwasser) [inline types/node-siege/index.d.ts:1]: Isn't this at version 0.2 at this point?

Comment (peterblazejewicz) [reply types/node-siege/index.d.ts:1]: oh, so there are two different packages with `siege` monicker (it tricked me also),
This one is also legit (node-siege) and bears `0.0.4` version:
https://github.com/npr/node-siege/blob/master/package.json#L4
It's kind of ol', but not deprecated, thx!

File: types/node-siege/index.d.ts (hunk 2 above)
Comment (Goldsmith42) [reply types/node-siege/index.d.ts:12]: Wrong README: https://www.npmjs.com/package/node-siege

File: types/node-siege/node-siege-tests.ts (hunk 3 above)
Comment (Goldsmith42) [reply types/node-siege/node-siege-tests.ts:13]: This is from a different package. These definitions are for this one: https://www.npmjs.com/package/node-siege

**Comment by Goldsmith42** [comment]
@DanielRosenwasser can you update your review? As far as I can see, all comments have to do with confusion between the actual package that these definitions are for ([node-siege](https://www.npmjs.com/package/node-siege)) and the similarly-named [siege](https://www.npmjs.com/package/siege) (which already has definitions here).

**Review by DanielRosenwasser — APPROVED** [review]
Sorry about the confusion

**Comment by Goldsmith42** [comment]
Ready to merge

**Pull request merged**